src/Microsoft.AspNet.OData.Shared/Formatter/EdmLibHelpers.cs

```diff
@@ -756,6 +756,12 @@ public static bool HasPrecision(EdmPrimitiveTypeKind primitiveTypeKind)
                     primitiveTypeKind == EdmPrimitiveTypeKind.TimeOfDay);
         }
 
+        public static IEdmTypeReference GetTypeReferenceOfProperty(IEdmModel model, Type clrType, string propertyName)
+        {
+            PropertyInfo propertyInfo = clrType.GetProperty(propertyName);
```

Comment:
This is a very expensive operation.  What are the scenarios where we call GetTypeReferenceOfProperty? is there another way to get the same information? Can the information be cached?

Reply:
We needed this only for enum types until ODL can start parsing enums without the type reference being passed.
Now, we append the value with 'enumType' to indicate the type and only in those cases try to obtain the typereference.